File: 04.Simulation模拟/leetcode_2553_separateDigits.py
from typing import List

class Solution:
    def separateDigits(self, nums: List[int]) -> List[int]:
        return list(map(int, list(''.join(list(map(str, nums))))))

if __name__ == "__main__":
    s = Solution()
    nums = [13,25,83,77]
    print(s.separateDigits(nums))

File: 04.Simulation模拟/leetcode_2553_separateDigits.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    vector<int> separateDigits(vector<int>& nums) {
        vector<int> res;
        for (int x : nums){
            string s = to_string(x);
            for (auto& c : s){
                res.push_back(c-'0');
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {12,25,83,7,7};
    vector<int> res = s.separateDigits(nums);
    for (int i : res){
        cout << i << " ";
    }
    cout << endl;
}
